framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -1253,6 +1347,32 @@ private V upperBoundOfValues(V otherVal, V thisVal, boolean shouldWiden) {
     return shouldWiden ? thisVal.widenUpperBound(otherVal) : thisVal.leastUpperBound(otherVal);
   }
 
+  /**
+   * Creates a new store that has all the values from both {@code this} and {@code other}. If a node
+   * has a value in both stores, then the most specific one is used.
```

Comment:
Why is the most specific one used rather than the glb?
It might be helpful to add a comment to the declaration of `mostSpecific` indicating when it should be called and when glb should be called.